File: j/java/src/main/java/sh/foxboy/api/listeners/MessageListener.java
package sh.foxboy.api.listeners;

import net.dv8tion.jda.api.events.message.MessageReceivedEvent;
import net.dv8tion.jda.api.hooks.ListenerAdapter;
import sh.foxboy.api.command.CommandMap;

public class MessageListener extends ListenerAdapter {

  private CommandMap commandMap;

  public MessageListener(CommandMap commandMap) {
    super();
    this.commandMap = commandMap;
  }

  @Override
  public void onMessageReceived(MessageReceivedEvent event) {
    commandMap.handleCommand(event.getMessage());
  }
}

File: k/kotlin/src/main/kotlin/sh/foxboy/api/listeners/MessageListener.kt
package sh.foxboy.api.listeners

import net.dv8tion.jda.api.events.message.MessageReceivedEvent
import net.dv8tion.jda.api.hooks.ListenerAdapter
import sh.foxboy.api.command.CommandMap


class MessageListener(private val commandMap: CommandMap) : ListenerAdapter() {

    override fun onMessageReceived(event: MessageReceivedEvent) {
        if (!event.message.author.isBot)
            println("Received message \"${event.message.contentRaw}\"")

        commandMap.handleCommand(event.message)
    }
}
